(hack) Add YAML plan 'prompt' step (#2715)
puppetlabs/bolt: documentation/writing_yaml_plans.md

      message: "The count is ${count}, and twice the count is ${double_count}"
```

### Prompt step

The `prompt` step displays a prompt to the user and waits for input.

Prompt steps support the following keys:

| Key | Type | Description | Required |
| --- | --- | --- | --- |

> | `default` | `String` | The default value to use if the user does not provide input or if standard in (stdin) is not a tty. | |

Reviewer: ```suggestion
| `default` | `String` | The default value to use if the user does not provide input or if standard input (stdin) is not a tty. | |
```
In this case, we should keep "standard input (stdin)"

> | `default` | `String` | The default value to use if the user does not provide input or if standard input (stdin) is not a tty. | |

Reviewer 2: This can be `Variant[Data, Target]` right? If that's what the keys of the menu hash accept.

Author: Updated to `Array`, `Boolean`, `Hash`, `Number`, `String`

| `menu` | `Array`, `Hash` | A list of options to choose from, or a hash of options to choose from where each key is the input used to select a value. | |
| `name` | `String` | The name of the variable to save the step result to. | |
| `prompt` | `String` | The prompt to display to the user. | ✓ |

> | `sensitive` | `Boolean` | Whether to disable echo back and mark the response as sensitive. The returned value will be wrapped by the Sensitive data type. | |

Reviewer: ```suggestion
| `sensitive` | `Boolean` | Whether to redact the response and mark it as sensitive. Bolt wraps sensitive values in the Sensitive data type. | |
```
I think this reads a little clearer. Not sure if there is a distinction here between echoing back to logs and the result. Let me know if this needs to be reworded further. This text appears elsewhere. I can open a separate ticket to fix if you accept this change.
